lib/next_ls/extensions/elixir_extension.ex

```diff
@@ -24,7 +24,7 @@ defmodule NextLS.ElixirExtension do
   end
 
   @impl GenServer
-  def handle_info({:compiler, diagnostics}, state) do
+  def handle_info({:compiler, diagnostics}, state) when is_list(diagnostics) do
```

Comment:
you'll need to handle an unhandled messaged, and emit a warning log

Reply:
Just pushed some changes/fixes for this branch. At this point, I don't think we need to handle unexpected messages here because this *should* always be a list. Prior to these changes, if there was an RPC call failure, this might be like a :nodedown message or something, which was then causing a more confusing downstream error.

At this point, this guard is basically just a safety net so that, if an invalid message is sent, it's more obvious/clear in the future.

Reply:
Forgot to add: I'm fine either removing this guard or adding a catch-all for unhandled messages, but I personally think it's okay to be assertive here.